### ISSUE TITLE
Users cannot delete their own accounts

### ISSUE THREAD

**Issue by georgemaschke** [RocketChat/Rocket.Chat#26696]
### Description: 

Users cannot delete their own accounts under Rocket.Chat 5.0.4

### Steps to reproduce:

1. As an unprivileged (non-admin) user, log into Rocket.Chat account;
2. Click on "My Account";
3. Choose "Delete my account";
4. Upon prompt, "If you are sure type in your username," type in user name and click "Delete" button;
5. A prompt saying "invalid password appears" (user was logged in, and there was never any indication that the password was required)
6. Account is not deleted.
7. Repeat process, but provide password and confirm it before clicking "Delete my account."
8. A prompt saying "invalid password appears" and account is not deleted.

### Expected behavior:

Account is deleted.

### Actual behavior:

Account is not deleted.

### Server Setup Information:

- Version of Rocket.Chat Server: 5.0.4
- Operating System: Ubuntu 20.04
- Deployment Method: snap
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: 14.19.3
- MongoDB Version: 5.0.6

### Client Setup Information

- Desktop App or Browser Version: Brave 1.42.97 and Tor Browser 
- Operating System: macOS 11.5.1

**Comment by ankar84** [comment]
duplicate of https://github.com/RocketChat/Rocket.Chat/issues/17927

**Comment by georgemaschke** [comment]
It's not a duplicate of #17927. That report concerns deletion of DMs, and the issue may be relevant to this one. However, this report concerns deletion of _accounts_.

**Comment by ankar84** [comment]
Sorry, my bad! You are right!

**Comment by Gummikavalier** [comment]
I confirm the deletion issue with locally created rocketchat accounts. Dialog asks for a username, but the password is required instead.

Deletion of accounts provisioned with Oauth works without issues.

**Comment by georgemaschke** [comment]
Entering the account password instead of the user name results in the account being deleted. The message presented after this is confusing, however. Instead of confirming account deletion, it unhelpfully states:

REGISTER USERNAME
The username is used to allow others to mention you in messages.
Loading suggestions

**Comment by Micheal-Earl** [comment]
> Entering the account password instead of the user name results in the account being deleted. The message presented after this is confusing, however. Instead of confirming account deletion, it unhelpfully states:
> 
> REGISTER USERNAME The username is used to allow others to mention you in messages. Loading suggestions

I was able to successfully delete my account by entering my password in the dialog box that asks for my username as per this post.

Issue still exists, dialog should ask for password instead of username (or maybe it is supposed to take username and it is mistakenly taking password?)

**Comment by PranavReddyG** [comment]
Yeah. I used password instead of the username and the account got deleted.

**Comment by hugocostadev** [comment]
Hi there, thanks for the contribution! 🚀 💯 

You are right, this still persists in the latest release.

The modal should display to enter the password with the password field

![image](https://user-images.githubusercontent.com/20212776/230361692-40a88b09-3c8a-4d91-9737-2018e785ed5a.png)


---

**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
- Feature Request: Open an issue in [Feature Request](https://github.com/RocketChat/feature-requests) repository

**Comment by Simer13** [comment]
@hugocostadev what is the ultimate changes that are required? 
Do you want the modal to ask for username or the password?

**Comment by hugocostadev** [comment]
> @hugocostadev what is the ultimate changes that are required? 
> 
> Do you want the modal to ask for username or the password?  

It should ask for password , and should be a password field instead of a text. The Backends already expects the password, it's just a matter to adjust the text and change the input type
